File: BoostDraftCodingAssignmentCSharp/SimpleXMLValidatorLibrary/SimpleXmlValidator.cs
using System;
using System.Collections.Generic;

namespace SimpleXMLValidatorLibrary
{
    public class SimpleXmlValidator
    {
        public static bool DetermineXml(string xml)
        {
            Stack<string> stack = new Stack<string>();
            int i = 0;

            while (i < xml.Length)
            {
                if (xml[i] == '<')
                {
                    int j = i + 1;

                    // Find the end of the tag
                    while (j < xml.Length && xml[j] != '>')
                    {
                        j++;
                    }

                    if (j == xml.Length)
                    {
                        return false; // Tag not closed properly
                    }

                    string tagContent = xml.Substring(i + 1, j - i - 1).Trim();

                    if (tagContent.StartsWith("/"))
                    {
                        // Closing tag
                        string tagName = tagContent.Substring(1).Trim();

                        if (stack.Count == 0 || stack.Peek() != tagName)
                        {
                            return false; // No matching opening tag
                        }

                        stack.Pop();
                    }
                    else
                    {
                        // Opening tag
                        string tagName = tagContent.Split(' ')[0].Trim();
                        stack.Push(tagName);
                    }

                    i = j;
                }

                i++;
            }

            return stack.Count == 0;
        }
    }
}

File: BoostDraftCodingAssignmentJava/src/xmlvalidatorlibrary/SimpleXmlValidator.java
package xmlvalidatorlibrary;

import java.util.Stack;

public class SimpleXmlValidator {

    public static boolean determineXml(String xml) {
        Stack<String> stack = new Stack<>();

        int i = 0;
        while (i < xml.length()) {
            if (xml.charAt(i) == '<') {
                int end = xml.indexOf('>', i);
                if (end == -1) {
                    return false; // Invalid XML if there is no closing '>'
                }

                String tag = xml.substring(i + 1, end);

                if (tag.startsWith("/")) {
                    // Closing tag
                    if (stack.isEmpty() || !stack.pop().equals(tag.substring(1))) {
                        return false; // Tags don't match
                    }
                } else {
                    // Opening tag
                    if (tag.contains(" ") || tag.contains("=")) {
                        return false; // Invalid if tag contains attributes or spaces
                    }
                    stack.push(tag);
                }

                i = end + 1;
            } else {
                i++;
            }
        }

        return stack.isEmpty(); // All tags should be properly closed
    }
}
